tests/FSharp.Test.Utilities/ILVerifierModule.fs

```diff
@@ -0,0 +1,74 @@
+        let errors = errors |> String.concat Environment.NewLine
+        errors, exitCode
+
+    let private verifyPEFileCore peverifierArgs (dllFilePath: string) =
+        let nuget_packages =
+            match Environment.GetEnvironmentVariable("NUGET_PACKAGES") with
+            | null ->
+                let profile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
+                $"""{profile}/.nuget/packages"""
+            | path -> path
+        let peverifyFullArgs = [ yield "exec"; yield $"""{nuget_packages}/dotnet-ilverify/9.0.0/tools/net9.0/any/ILVerify.dll"""; yield "--verbose"; yield dllFilePath; yield! peverifierArgs ]
+        let workingDirectory = Path.GetDirectoryName dllFilePath
+        let _, exitCode =
+            let peverifierCommandPath = Path.ChangeExtension(dllFilePath, ".peverifierCommandPath.cmd")
+            let args = peverifyFullArgs |> Seq.fold(fun a acc -> $"{a} " + acc) ""
+            File.WriteAllLines(peverifierCommandPath, [| $"{args}" |] )
+            File.Copy(typeof<RequireQualifiedAccessAttribute>.Assembly.Location, Path.GetDirectoryName(dllFilePath) ++ "FSharp.Core.dll", true)
+            exec config.DotNetExe peverifyFullArgs workingDirectory
+
+        // Grab output
+        let outputText = File.ReadAllText(Path.Combine(workingDirectory, "StandardOutput.txt"))
+        let errorText = File.ReadAllText(Path.Combine(workingDirectory, "StandardError.txt"))
+
+        match exitCode with
+        | 0 -> {Outcome = NoExitCode; StdOut = outputText; StdErr = errorText } 
+        | _ -> {Outcome = ExitCode exitCode; StdOut = outputText; StdErr = errorText } 
+
+    let private verifyPEFileAux (compilationResult: CompilationResult) args =
+        let result =
+            match compilationResult.Compilation with
+            | FS _ ->
+                match compilationResult, compilationResult.OutputPath with
+                | CompilationResult.Success result, Some name ->
+                    let verifyResult = verifyPEFileCore args name
+                    match verifyResult.Outcome with
+                    | NoExitCode -> CompilationResult.Success {result with Output = Some (ExecutionOutput verifyResult)}
+                    | ExitCode _ ->  CompilationResult.Failure {result with Output = Some (ExecutionOutput verifyResult)}
+                    | failed -> failwith $"Compilation must succeed in order to verify IL.{failed}"
+                | failed ->
+                    failwith $"""Compilation must succeed in order to verify IL.{failed}"""
+            | _ ->
+                failwith "PEVerify is only supported for F#."
+        result
+
+    let verifyPEFile compilationResult =
+        verifyPEFileAux compilationResult [||]
+
+    let verifyPEFileWithArgs compilationResult args =
+        verifyPEFileCore compilationResult args
+
+    let verifyPEFileWithSystemDlls compilationResult =
```

Comment:
The full set of args created here will be static within the same process - the args string can be stored and retrieved just once.

Reply:
done